script/cache_critical_dns

```diff
@@ -260,6 +263,16 @@ HEALTH_CHECKS = {
     redis_healthcheck(
       host: addr,
       password: ENV["DISCOURSE_REDIS_PASSWORD"])},
+  "DISCOURSE_MESSAGE_BUS_REDIS_HOST": lambda { |addr|
+    redis_healthcheck(
+      host: addr,
+      port: ENV["DISCOURSE_MESSAGE_BUS_REDIS_PORT"],
```

Comment:
If `DISCOURSE_MESSAGE_BUS_REDIS_PORT` is not set in the environment, `redis_healthcheck` will assign the value of `nil` to the `client_opts` hash.  The default value of `6379` appears to be unused in all `MESSAGE_BUS` cases.  We should do one of the following:

* Hard-fail if `REDIS_HOST` is set without `REDIS_PORT`.  (Message Bus only.)
* Actually use the default of `6379`.

The `port` method parameter (what do you call these in Ruby?) defaults to a value of integer type, however this expression will evaluate to a value of string type.  I guess the Ruby library does not care, but this may be confusing.

Because I am Ruby-challenged, here is how I might do it in Python:

```python
#!/usr/bin/env python3

# To see this in action,
# try executing with and without REDIS_PORT in the environment.
#
#   % ./hello
#   host='127.0.0.1'
#   port=6379
#   password='open-sesame'
#
#   % REDIS_PORT=12345 ./hello
#   host='127.0.0.1'
#   port=12345
#   password='open-sesame'

import os

def redis_healthcheck(host, port=6379, password=None):
    # implementation goes here
    print(f"{host=}")
    print(f"{port=}")
    print(f"{password=}")


def get_as_int(obj, name, default=None):
    val = obj.get(name, default)
    if val is None:
        return val
    return int(val)


redis_healthcheck("127.0.0.1", **({
    "password": "open-sesame",
    **({"port": v} if (v := get_as_int(os.environ, "REDIS_PORT")) else {})}))
```

Reply:
I've made some changes as suggested. Turns out a `env_as_int` method was useful in more than one place.